Server/mods/deathmatch/logic/CPlayerManager.cpp

```diff
@@ -320,12 +320,15 @@ void CPlayerManager::Broadcast(const CPacket& Packet, const std::multimap<ushort
     DoBroadcast(Packet, groupMap);
 }
 
+const bool CPlayerManager::ValidPlayerModels[313] = {1,1,1,0,0,0,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1};
+
 bool CPlayerManager::IsValidPlayerModel(unsigned short usPlayerModel)
 {
-    return (usPlayerModel == 0 || usPlayerModel == 1 || usPlayerModel == 2 || usPlayerModel == 7 ||
-            (usPlayerModel >= 9 && usPlayerModel != 208 && usPlayerModel != 149 && usPlayerModel != 119 && usPlayerModel != 86 && usPlayerModel != 74 &&
-             usPlayerModel != 65 && usPlayerModel != 42 && usPlayerModel <= 272) ||
-            (usPlayerModel >= 274 && usPlayerModel <= 288) || (usPlayerModel >= 290 && usPlayerModel <= 312));
+    if (usPlayerModel < 0 || usPlayerModel > 312)
+    {
+        return false;
+    }
+    return (ValidPlayerModels[usPlayerModel]);
 }
```

Comment:
```suggestion
bool CPlayerManager::IsValidPlayerModel(unsigned short model)
{
    if (model > 312)
        return false; // TODO: On client side maybe check if a model was allocated with engineRequestModel and it is a ped
        
    switch (model)
    {
        case 3:
        case 4:
        case 5:
        case 6:
        case 8:
        case 42:
        case 65:
        case 74:
        case 86:
        case 119:
        case 149:
        case 208:
        case 273:
        case 289:
            return false;
        default:
            return true;
    }
}
```

I think this solution is the cleanest. 
We could add a function that returns an array (on server side a static one (eg.: `std::array<unsigned short, 300>`), on client we'd ideally also include the engineRequestModel ones as well).

Reply:
I also prefer this for readability